### ISSUE TITLE
Problem with BuildPackages in 32-bit builds

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#1126]
On a Jenkins slave, I call `../bin/BuildPackages.sh` and it picks up that the GAP kernel is built in 32-bit mode:

```
Using GAP location: /data/gap-jenkins/workspace/GAP-major-release-test/GAPCOPTS/32build/GAPTARGET/standard/label/kovacs/gap4r9
Building with 32-bit ABI
Attempting to build GAP packages.
Note that many GAP packages require extra programs to be installed,
and some are quite difficult to build. Please read the documentation for
packages which fail to build correctly, and only worry about packages
you require!
Logging into ./log/buildpackages.log
...
```
but then it fails on a number of packages, for example,
```
==== Checking anupq-3.1.4
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking for gawk... (cached) gawk
checking for gcc... gcc
checking whether the C compiler works... no
ERROR: configure: error: in `/data/gap-jenkins/workspace/GAP-major-release-test/GAPCOPTS/32build/GAPTARGET/standard/label/kovacs/gap4r9/pkg/anupq-3.1.4':
ERROR: configure: error: C compiler cannot create executables
ERROR: See `config.log' for more details

WARNING: Failed to build anupq-3.1.4
```
and
```
==== Checking io-4.4.6
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether make supports nested variables... (cached) yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking for gcc... gcc
checking whether the C compiler works... no
ERROR: configure: error: in `/data/gap-jenkins/workspace/GAP-major-release-test/GAPCOPTS/32build/GAPTARGET/standard/label/kovacs/gap4r9/pkg/io-4.4.6':
ERROR: configure: error: C compiler cannot create executables
ERROR: See `config.log' for more details
ERROR: gmake: *** No targets specified and no makefile found.  Stop.

WARNING: Failed to build io-4.4.6
```
It builds, however, some others, e.g. kbmag.

**Comment by ChrisJefferson** [comment]
I think the fix for this is this patch, but could you try it on the jenkins machine, rather than it get pushed in a circle?

```
diff --git a/bin/BuildPackages.sh b/bin/BuildPackages.sh
index 76dd2b4a5..1f9f1458b 100755
--- a/bin/BuildPackages.sh
+++ b/bin/BuildPackages.sh
@@ -200,7 +200,7 @@ run_configure_and_make() {
   then
     if grep Autoconf ./configure > /dev/null
     then
-      ./configure "$CONFIGFLAGS" --with-gaproot="$GAPDIR"
+      ./configure $CONFIGFLAGS --with-gaproot="$GAPDIR"
     else
       ./configure "$GAPDIR"
     fi
```

**Comment by alex-konovalov** [comment]
Thanks @ChrisJefferson - seems to work. Will you make a PR?

**Comment by fingolfin** [comment]
This is addressed by #1128 -- it includes a change equivalent to what @ChrisJefferson proposes, though I arrived at it for a different reason -- I noticed that configure scripts would print weird messages about --build and --host and stuff... Then after I realized the cause, I thought "wait, there was also a bug report about 32bit issues" and it made "click" ;-).

**Comment by fingolfin** [comment]
Fixed by #1128.
